src/object/cli_shard.c

```diff
@@ -1886,6 +1886,8 @@ dc_obj_shard_list(struct dc_obj_shard *obj_shard, enum obj_rpc_opc opc,
 		oei->oei_epr.epr_hi = args->la_auxi.epoch.oe_value;
 		oei->oei_flags |= ORF_ENUM_WITHOUT_EPR;
 	}
+	if (!obj_args->incr_order)
+		oei->oei_flags |= ORF_DESCENDING_ORDER;
```

Comment:
only need this for recx list?

Reply:
We also check oei_flags in src/object/srv_obj.c and set "VOS_IT_RECX_REVERSE" if applicable.

Reply:
oh, yes. It won't cause any issue for now.

But only recx list(daos_obj_list_recx) have incr_order option, list akey/dkey/obj does not have this option, but daos_obj_list_t are shared by all these api, so better to limit this check for recx list, in case it will cause issue in the future, IMHO.

If you need upgrade the patch, then please fix this, otherwise I am fine to update it later.

Reply:
It seems that dc_obj_list_obj_task_create has argument incr_order too. I just revised the code to apply this flag to only DAOS_OBJ_RPC_ENUMERATE and DAOS_OBJ_RECX_RPC_ENUMERATE. Please let me know if we do not want to apply it DAOS_OBJ_RPC_ENUMERATE. Thank you very much!